### ISSUE TITLE
node-properties are not parsing correctly

### ISSUE THREAD

**Issue by gcentauri** [200ok-ch/org-parser#6]
From trying to reproduce https://github.com/200ok-ch/organice/blob/master/src/lib/org_utils.unit.test.js i discovered that the parsing of drawers is not complete. For example, the output of the following org:

```
* Header 2
:PROPERTIES:
:foo2: abc
:bar: zyx
:baz: test
:emptyprop:
:END:
```

is this:

``` clojure
 [:drawer-begin-line [:drawer-name "PROPERTIES"]]
 [:content-line ":foo2: abc"]
 [:content-line ":bar: zyx"]
 [:content-line ":baz: test"]
 [:drawer-begin-line [:drawer-name "emptyprop"]]
 [:drawer-begin-line [:drawer-name "END"]])
```

The properties of the drawer are just being parsed as content-lines, and the "emptyprop"
is seen as another =:drawer-begin-line=. I assume this is supposed to produce a nested
structure something like this:

``` clojure
[:drawer-begin-line 
   [:drawer-name "PROPERTIES"]
   [:property-line [:node-property-name "foo2"] [:node-property-value "abc"]]
   [:property-line [:node-property-name "bar"] [:node-property-value "zyx"]]
   [:property-line [:node-property-name "baz"] [:node-property-value "test"]]
   [:property-line [:node-property-name "emptyprop"] [:node-property-value ""]]
   [:drawer-end-line]
 ]
```
Org's own built in parser builds a structure like this:

```emacs-lisp
   (property-drawer
    (:begin 117 :end 181 :contents-begin 130 :contents-end 174 :post-blank 1 :post-affiliated 117 :parent #2)
    (node-property (:key "foo2" :value "abc" :begin 130 :end 141 :post-blank 0 :post-affiliated 130 :parent #3))
    (node-property (:key "bar" :value "zyx" :begin 141 :end 151 :post-blank 0 :post-affiliated 141 :parent #3))
    (node-property (:key "baz" :value "test" :begin 151 :end 162 :post-blank 0 :post-affiliated 151 :parent #3))
    (node-property (:key "emptyprop" :value "" :begin 162 :end 174 :post-blank 0 :post-affiliated 162 :parent #3)))
```

**Comment by gcentauri** [comment]
I'd like to work on this issue, but I've run out of time for the day. I thought I'd post my findings here in case someone has suggestions or direction. I've just started looking at the EBNF specification to see what might be incorrect or missing.

**Comment by gcentauri** [comment]
turns out i had a little more time, part of this problem is solved by adding `node-property-line` right before `content-line` in the definition of `<line>` in `org.ebnf`.  Its a bit closer now

```clojure
[:drawer-begin-line [:drawer-name "PROPERTIES"]]
 [:node-property-line
  [:node-property-name "foo2"]
  [:node-property-value "abc"]]
 [:node-property-line [:node-property-name "bar"] [:node-property-value "zyx"]]
 [:node-property-line
  [:node-property-name "baz"]
  [:node-property-value "test"]]
 [:drawer-begin-line [:drawer-name "emptyprop"]]
 [:drawer-begin-line [:drawer-name "END"]]]
```

**Comment by munen** [comment]
Sweet, good catch! I'm happy to see that the organice tests for Org do help a little in building out org-parser^^

**Comment by schoettl** [comment]
I will work on this in #11 – I'll try to parse the property drawer as one thing instead of line-based. The resulting structure should then be something like this:
```
[:drawer-begin-line [:drawer-name "PROPERTIES"] [:nodes
 [:node-property-line
  [:node-property-name "foo2"]
  [:node-property-value "abc"]]
 [:node-property-line [:node-property-name "bar"] [:node-property-value "zyx"]]
 [:node-property-line
  [:node-property-name "baz"]
  [:node-property-value "test"]]]
;; (no END line necessary)
```

Regarding the misread "emptyprop": Haven't checked, but maybe there *must* be a space after `:propname:` even for empty props. But anyway, it can't open a new drawer within another drawer.

**Comment by munen** [comment]
Closed by #25.
